luigi/contrib/hadoop_jar.py

```diff
@@ -52,7 +52,10 @@ def fix_paths(job):
                 logger.info('Using temp path: %s for path %s', y.path, x.path)
                 args.append(y.path)
         else:
-            args.append(str(x))
+            try:
+                args.append(x.path)
```

Comment:
Can you add a comment to this `try-except` dance? Perhaps just linking back to this exact PR.